tests/src/main/scala/scalaz/tests/IOTests.scala

```diff
@@ -0,0 +1,107 @@
+package scalaz
+package tests
+
+import scala.{ List, Range, StringContext }
+import scala.Predef.String
+
+import Scalaz._
+
+import testz._, z._
+import laws._, FunctorLaws._
+
+import zio._
+
+object IOTests {
+
+  case class Error(msg: String)
+
+  // a probably-too-large selection of IO actions; more could be added
+  val ios = {
+    type IOES = IO[Error, String]
+    type Endo = IOES => IOES
+    val seeds = List[IOES](
+      IO.now("now"),
+      IO.unit.map(_ => "unit"),
+      IO.point("point"),
+      IO.fail(Error("error")),
+      //IO.sleep(scala.concurrent.duration.Duration.fromNanos(1)).map(_ => "slept")
+    )
+    val endos0: List[Endo] =
+      seeds.map(s => (io: IOES) => io.flatMap(_ => s)) :::
+        List(fns.s1, fns.s2, fns.f).map(f => (io: IOES) => io.flatMap(f)) :::
+        List[Endo](
+        _.map(_ + "ishly"),
+      )
+    val endos = Range(1, endos0.size)
+      .flatMap(endos0.combinations)
+      .flatMap(l => l)
+    seeds.flatMap(s => endos.map(_(s)))
+  }
+
+  object RTS extends RTS
+
+  def ioEql[E, A](l: IO[E, A], r: IO[E, A]): Result =
+    assert(RTS.unsafeRunSync(l.attempt) == RTS.unsafeRunSync(r.attempt))
+
+  object fns {
```

Comment:
Why not just define this as a list of `fns`?

Reply:
I want to be able to do all pairs of success/failure in the bind associativity test